filestore/test/test_retrieve.py

```diff
@@ -106,24 +104,23 @@ def test_register_fail():
         # the same class with the same name
         fsa.register_handler('syn-mod', SynHandlerMod)
         # should raise as it is trying to change the registered class
-        assert_raises(RuntimeError, fsa.register_handler,
-                      'syn-mod', SynHandlerEcho)
+        with pytest.raises(RuntimeError):
+            fsa.register_handler('syn-mod', SynHandlerEcho)
 
 
 def test_context_manager_replace():
     with fsa.handler_context({'syn-mod': SynHandlerMod}):
-        assert_true(fsa._FS_SINGLETON.handler_reg['syn-mod'] is SynHandlerMod)
+        assert fsa._FS_SINGLETON.handler_reg['syn-mod'] is SynHandlerMod
         with fsa.handler_context({'syn-mod': SynHandlerEcho}):
-            assert_true(fsa._FS_SINGLETON.handler_reg['syn-mod']
-                        is SynHandlerEcho)
-        assert_true(fsa._FS_SINGLETON.handler_reg['syn-mod'] is SynHandlerMod)
-    assert_false('syn-mod' in fsa._FS_SINGLETON.handler_reg)
+            assert fsa._FS_SINGLETON.handler_reg['syn-mod'] is SynHandlerEcho
+        assert fsa._FS_SINGLETON.handler_reg['syn-mod'] is SynHandlerMod
+    assert 'syn-mod' in fsa._FS_SINGLETON.handler_reg
```

Comment:
@tacaswell this is now failing.  If I inspect `fsa._FS_SINGLETON.handler_reg` before the `with` statement it is empty:

```
> /home/edill/dev/python/filestore/filestore/test/test_retrieve.py(113)test_context_manager_replace()
-> with fsa.handler_context({'syn-mod': SynHandlerMod}):
(Pdb) fsa._FS_SINGLETON.handler_reg
ChainMap({})
```

What are the setup conditions for this test function?  I would imagine there is supposed to be a `syn-mod` handler in `fsa._FS_SINGLETON.handler_reg` before this test function gets executed?